posthog/hogql/resolver.py

```diff
@@ -306,7 +306,7 @@ def visit_join_expr(self, node: ast.JoinExpr):
             if isinstance(database_table, SavedQuery):
                 self.current_view_depth += 1
 
-                if self.current_view_depth > self.context.max_view_depth:
+                if self.context.max_view_depth is not None and self.current_view_depth > self.context.max_view_depth:
```

Comment:
If we're supporting these now should just completely remove. May end up with some heavy queries temporarily. Also related to handling cycles comment below

Reply:
Happy to remove, I think I was just being cautious in case there was some use case to limiting query depth _sometimes_.

Reply:
Main reason was just performance so no other gotchas